bokeh/model.py

```diff
@@ -458,19 +458,21 @@ def js_link(self, attr, other, other_attr, attr_selector=None):
                 )
 
         '''
-        if attr not in self.properties():
+        descriptor = self.lookup_descriptor(attr)
+        if descriptor is None:
```

Comment:
Is this an optimization to avoid calling `accumulate_from_superclasses` ?  A we certain it behaves identically?

Reply:
No, the point is to work (at least internally) with symbolic descriptors (objects) instead of strings, which not only is more robust on its own, it also makes implementing aliases trivial. This is how bokehjs handles things as well. Honestly speaking, I don't know how this relates to any of the "accumulate" logic of bokeh. I assume this to have the same resolution rules as plain Python.

Reply:
just informational:

IIRC Bokeh properties were loosely based on the Traits implementation, except in pure python. As best I can remember after a quick look, the accumulate logic is a complicated caching and partitioning system. you can see what `.properties()` does:
```
        if with_bases:
            return accumulate_from_superclasses(cls, "__properties__")
        else:
            return set(cls.__properties__)
```
It allows:
* searching *only* the properties (contained strictly per class in `__properties__` set up by the meta class
* searching only on *this class* (excluding all base classes). 

However it's probably worth investigating whether some or all of this can be simplified at some point:

* can we search based on type checking or tagging to find properties instead of manufacturing `__properties__`, `__properties_with_refs__`, etc?
* do we need the "with_bases" search partitioning capability at all? 

Assuming looking up things using the standard name resolution is not some huge performance drag then I think a lot of `MetaHasProps` machinery could potentially be streamlined (which would be great).

Reply:
Tho thinking a bit more I expect some of the class partitioning machinery is to be able to prevent properties from being re-declared one sub-classes. We do need to make sure that can continue, regardless. 

Again, this is just FYI, nothing to do here. I am willing to believe *general lookup* is going to behave identically to standard name resolution, and quick check seems (running on this branch) to indicate it is only negligibly slower:
```
In [6]: %timeit p.lookup("border_fill_color")
582 ns ± 10.7 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [7]: %timeit "border_fill_color" in p.properties()
426 ns ± 12.5 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

Reply:
issue for the future here:  https://github.com/bokeh/bokeh/issues/10867